sfn/src/main/scala/uk/gov/nationalarchives/DASFNClient.scala

```diff
@@ -0,0 +1,61 @@
+package uk.gov.nationalarchives
+
+import cats.effect.Async
+import io.circe.{Encoder, Printer}
+import io.circe.syntax._
+import software.amazon.awssdk.http.async.SdkAsyncHttpClient
+import software.amazon.awssdk.http.nio.netty.NettyNioAsyncHttpClient
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.sfn.SfnAsyncClient
+import software.amazon.awssdk.services.sfn.model.{StartExecutionRequest, StartExecutionResponse}
+
+/** An SFN client. It is written generically so can be used for any effect which has an Async instance. Requires an
+  * implicit instance of cats Async which is used to convert CompletableFuture to F
+  *
+  * @param sfnAsyncClient
+  *   An AWS SFN Async client
+  * @tparam F
+  *   Type of the effect
+  */
+class DASFNClient[F[_]: Async](sfnAsyncClient: SfnAsyncClient) {
+
+  /** @param stateMachineArn
+    *   The arn of the state machine to start
+    * @param input
+    *   A case class. This will be deserialised to a json string and sent as input to the step function.
+    * @param name
+    *   An optional name. If this is omitted, AWS will generate a UUID for a name.
```

Comment:
I was thinking maybe either this parameter could have a more specific name like `sfnName` or the description instead  reads "An optional step function name."?

Reply:
I'll leave it called `name` because it's called that in the AWS classes. I've updated the description.